router-tests/cors_test.go

```diff
@@ -0,0 +1,205 @@
+package integration
```

Comment:
Perhaps a test for max recursion depth? Change the test instance to 2 or so and check that it fails?

Reply:
good point! Added